File: backend/src/main/java/com/thirdhand/campusvault/service/BookingService.java
package com.thirdhand.campusvault.service;

import com.thirdhand.campusvault.dto.request.CreateBookingRequest;
import com.thirdhand.campusvault.dto.response.BookingResponse;

import java.util.List;

public interface BookingService {

    BookingResponse createBooking(CreateBookingRequest request);

    BookingResponse getBookingById(Long bookingId);

    List<BookingResponse> getAllBookings();

    BookingResponse approveBooking(Long bookingId);

    BookingResponse rejectBooking(Long bookingId);

    BookingResponse cancelBooking(Long bookingId);

    BookingResponse completeBooking(Long bookingId);
}

File: frontend/web/lib/services/bookingService.ts
import api from "../api";
import { BookingResponse } from "../../types/booking";

export const bookingService = {
  getAllBookings: async (): Promise<BookingResponse[]> => {
    const response = await api.get<BookingResponse[]>("/bookings");
    return response.data;
  },

  getMyBookings: async (): Promise<BookingResponse[]> => {
    const response = await api.get<BookingResponse[]>("/bookings/me");
    return response.data;
  },

  getRequestsForMyListings: async (): Promise<BookingResponse[]> => {
    const response = await api.get<BookingResponse[]>("/bookings/requests");
    return response.data;
  },

  createBooking: async (bookingData: {
    itemId: number;
    startDate: string;
    endDate: string;
  }): Promise<BookingResponse> => {
    const response = await api.post<BookingResponse>("/bookings", bookingData);
    return response.data;
  },

  approveBooking: async (bookingId: number): Promise<BookingResponse> => {
    const response = await api.post<BookingResponse>(`/bookings/${bookingId}/approve`);
    return response.data;
  },

  rejectBooking: async (bookingId: number): Promise<BookingResponse> => {
    const response = await api.post<BookingResponse>(`/bookings/${bookingId}/reject`);
    return response.data;
  },

  cancelBooking: async (bookingId: number): Promise<BookingResponse> => {
    const response = await api.post<BookingResponse>(`/bookings/${bookingId}/cancel`);
    return response.data;
  },

  completeBooking: async (bookingId: number): Promise<BookingResponse> => {
    const response = await api.post<BookingResponse>(`/bookings/${bookingId}/complete`);
    return response.data;
  },
};
